sql/core/src/main/scala/org/apache/spark/sql/json/JsonRDD.scala

```diff
@@ -48,7 +48,11 @@ private[sql] object JsonRDD extends Logging {
     require(samplingRatio > 0, s"samplingRatio ($samplingRatio) should be greater than 0")
     val schemaData = if (samplingRatio > 0.99) json else json.sample(false, samplingRatio, 1)
     val allKeys =
-      parseJson(schemaData, columnNameOfCorruptRecords).map(allKeysWithValueTypes).reduce(_ ++ _)
+      if (schemaData.isEmpty()) {
+        Set[(String,DataType)]()
```

Comment:
This is a super nit, but I think typically we would just do `Set.empty` here or maybe `Set.empty[(String, DataType)]` if you really want to be explicit.